invenio_vocabularies/datastreams/readers.py

```diff
@@ -87,4 +90,58 @@ def read(self):
                 # todo add logging/fail
                 pass
 
-            yield StreamEntry(resp.content)
+            yield resp.content
+
+
+class ZipReader(BaseReader):
+    """ZIP reader."""
+
+    def __init__(self, *args, options=None, regex=None, **kwargs):
+        """Constructor."""
+        self._options = options or {}
+        self._regex = re.compile(regex) if regex else None
+        super().__init__(*args, **kwargs)
+
+    def read(self, item=None):
+        """Opens a ZIP and iterates through the files in the archive."""
+        # https://docs.python.org/3/library/zipfile.html
+        filepath = item or self._origin
+        with zipfile.ZipFile(filepath, **self._options) as archive:
+            for member in archive.infolist():
+                match = not self._regex or self._regex.search(member.filename)
+                if not member.is_dir() and match:
+                    yield archive.open(member)
+
+
+class JsonReader(BaseReader):
+    """JSON object reader."""
+
+    def read(self, item=None, **kwargs):
+        """Reads (loads) a json object and yields its items."""
+        entries = json.load(item)
```

Comment:
To be consistent with other readers
```suggestion
        filepath = item or self._origin
        entries = json.load(filepath)
```

Reply:
Fixed